yarn-project/noir-protocol-circuits/src/scripts/generate_ts_from_abi.ts

```diff
@@ -209,11 +226,20 @@ const circuits = [
 ];
 
 const main = async () => {
+  try {
+    await fs.access('./src/types/');
+  } catch (error) {
+    await fs.mkdir('./src/types', { recursive: true });
+  }
+
   for (const circuit of circuits) {
     const rawData = await fs.readFile(`./src/target/${circuit}.json`, 'utf-8');
     const abiObj: NoirCompiledCircuit = JSON.parse(rawData);
     const generatedInterface = generateTsInterface(abiObj.abi);
-    await fs.writeFile(`./src/types/${circuit}_types.ts`, generatedInterface);
+
+    const outputFile = `./src/types/${circuit}_types.ts`;
+    await fs.writeFile(outputFile, generatedInterface);
+    runLintAndFormat(outputFile);
```

Comment:
Why move the formatting to this file?

Reply:
it would fail when I tried to run `formatting:fix` in its entirety at that point of the docker image, due to missing dependency @aztec/merkle-tree in one of the test files